sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -301,6 +302,16 @@ object CatalystTypeConverters {
       DateTimeUtils.toJavaTimestamp(row.getLong(column))
   }
 
+  private object IntervalConverter extends CatalystTypeConverter[Interval, Interval, Array[Byte]] {
+    override def toCatalystImpl(scalaValue: Interval): Array[Byte] =
```

Comment:
But If we use `Interval` as internal type, maybe we should just use a int and a long not 12 bytes? It looks to me that keeping an object with int + long or 12 bytes makes no difference, as object itself has a lot of overhead.